core/app/models/spree/variant.rb

```diff
@@ -30,7 +30,7 @@ class Variant < Spree::Base
     attr_writer :rebuild_vat_prices
     include Spree::DefaultPrice
 
-    belongs_to :product, -> { with_discarded }, touch: true, class_name: 'Spree::Product', inverse_of: :variants, optional: false
+    belongs_to :product, -> { with_discarded }, touch: true, class_name: 'Spree::Product', inverse_of: :variants_including_master, optional: false
```

Comment:
It seems to me that this change (and the one in product.rb) are not really related to the issue with `ActiveRecord.has_many_inverse`, but they've been an issue for a while. 

What about moving these changes into their own commit? I think it would give the fix more prominence and make the last commit a bit lighter.

Reply:
Ops, I merged too fast. Sorry, @spaghetticode! 🤦‍♂️  

I would have agreed with you for what it's worth. Maybe something we can take into account in the next PRs. Sorry again.

Reply:
Fixing that association solved two test failures:

https://github.com/solidusio/solidus/blob/69b9a0f74c850d1283f24854332036cfa69db071/api/spec/requests/spree/api/products_controller_spec.rb#L251

```
  1) Spree::Api::ProductsController as an admin creating a product creates with embedded variants
     Failure/Error: expect(variants.count).to eq(2)
     
       expected: 2
            got: 3
     
       (compared using ==)
```

https://github.com/solidusio/solidus/blob/69b9a0f74c850d1283f24854332036cfa69db071/core/spec/lib/spree/core/testing_support/factories/variant_factory_spec.rb#L25

```
  1) variant factory master variant builds a master variant properly
     Failure/Error: expect(product.variants).to be_empty
       expected `#<ActiveRecord::Associations::CollectionProxy []>.empty?` to be truthy, got false
```

I didn't debug further, though, as I saw that the change made sense.